File: examples/macro-briefing.mjs
import { pathToFileURL } from 'node:url';

export async function fetchSeries(id, fetcher = fetch) {
  if (!/^[A-Z0-9:_-]{1,128}$/.test(id)) throw new Error('Invalid series ID');
  const response = await fetcher('https://api.meridianapi.io/v1/series/' + encodeURIComponent(id), {
    headers: { Accept: 'application/json' }, signal: AbortSignal.timeout(15000),
  });
  if (!response.ok) throw new Error('Meridian returned HTTP ' + response.status);
  const series = await response.json();
  if (series?.id !== id || typeof series.units !== 'string' || !Array.isArray(series.observations)) throw new Error('Invalid series response');
  const latest = series.observations.filter(point => typeof point.date === 'string' && Number.isFinite(point.value)).sort((a,b) => a.date.localeCompare(b.date)).at(-1);
  if (!latest) throw new Error('No observations for ' + id);
  return { id, source: series.source, units: series.units, ...latest };
}

if (process.argv[1] && import.meta.url === pathToFileURL(process.argv[1]).href) {
  try {
    for (const id of ['UNRATE', 'CPIAUCSL', 'FEDFUNDS']) console.log(JSON.stringify(await fetchSeries(id)));
  } catch (error) {
    console.error(error instanceof Error ? error.message : 'Request failed');
    process.exitCode = 1;
  }
}

File: examples/macro_briefing.py
import json
import sys
from urllib.error import HTTPError, URLError
from urllib.request import Request, urlopen


def main():
    for series_id in ("UNRATE", "CPIAUCSL", "FEDFUNDS"):
        request = Request(
            "https://api.meridianapi.io/v1/series/" + series_id,
            headers={"Accept": "application/json"},
        )
        with urlopen(request, timeout=15) as response:
            series = json.load(response)
        observations = [
            point for point in series["observations"]
            if point.get("value") is not None
        ]
        if not observations:
            raise ValueError("No observations for " + series_id)
        latest = max(observations, key=lambda point: point["date"])
        print(json.dumps({
            "id": series_id, "source": series["source"],
            "units": series["units"], **latest,
        }))


if __name__ == "__main__":
    try:
        main()
    except (HTTPError, URLError, TimeoutError, ValueError, KeyError) as error:
        print("Meridian request failed: " + str(error), file=sys.stderr)
        sys.exit(1)
